interpreter/cling/lib/Interpreter/Interpreter.cpp

```diff
@@ -244,6 +261,43 @@ namespace cling {
     bool usingCxxModules = getSema().getLangOpts().Modules;
 
     if (usingCxxModules) {
+      HeaderSearch& HSearch = getCI()->getPreprocessor().getHeaderSearchInfo();
+
+      // Get system include paths
+      llvm::SmallVector<std::string, 3> HSearchPaths;
+      for (auto Path = HSearch.system_dir_begin(); Path < HSearch.system_dir_end(); Path++) {
+        HSearchPaths.push_back((*Path).getName());
+      }
+
+      // Virtual modulemap overlay file
+      std::string MOverlay = "{\n 'version': 0,\n 'roots': [\n";
+
+      // Check if the system path exists. If it does and it contains "gcc" (as stl path is always inferred from
+      // gcc path), append this to MOverlay.
+      struct stat BufferTmp;
+      for (auto SystemPath : HSearchPaths) {
+        if ((::stat(SystemPath.c_str(), &BufferTmp) == 0) && (SystemPath.find("gcc") != std::string::npos
```

Comment:
I'm assuming that the path of STL headers contains "gcc" or "c++", this is fragile and I would like to do something like `g++ -print-search-dirs` to get the information (HSearchPaths contain other system paths like "/usr/local/lib/cmake/VecCore/../../../include" which I don't want to overlay with stl.modulemap). Do you know how to get STL header path information from C++ compiler? Otherwise I will just leave FIXME here..